server/interpreter/testcases.js

```diff
@@ -0,0 +1,460 @@
+    }
+    else {
+        "else";
+    }
+    `,
+    expected: "then" },
+
+  { name: "ifFalse", src: `
+    if(false) {
+        "then";
+    }
+    else {
+        "else";
+    }
+    `,
+    expected: "else" },
+
+  { name: "simpleAssignment", src: `
+    var x = 0;
+    x = 44;
+    x;
+    `,
+    expected: 44 },
+
+  { name: "propertyAssignment", src: `
+    var o = {};
+    o.foo = 45;
+    o.foo;
+    `,
+    expected: 45 },
+
+  { name: "postincrement", src: `
+    var x = 45;
+    x++;
+    x++;
+    `,
+    expected: 46 },
+
+  { name: "preincrement", src: `
+    var x = 45;
+    ++x;
+    ++x;
+    `,
+    expected: 47 },
+
+  { name: "concat", src: `
+    "foo"+"bar";
+    `,
+    expected: "foobar" },
+
+  { name: "plusequalsLeft", src: `
+    var x=40, y=8;
+    x+=y;
+    x;
+    `,
+  expected: 48 },
+
+  { name: "plusequalsRight", src: `
+    var x=40, y=8;
+    x+=y;
+    y;
+    `,
+    expected: 8 },
+
+  { name: "simpleFunctionExpression", src: `
+    var v;
+    var f = function() {v = 49};
+    f();
+    v;
+    `,
+    expected: 49 },
+
+  { name: "fExpWithParameter", src: `
+    var v;
+    var f = function(x) {v = x};
+    f(50);
+    v;
+    `,
+    expected: 50 },
+
+  { name: "functionWithReturn", src: `
+    (function(x) {return x})(51);
+    `,
+    expected: 51 },
+
+  { name: "functionWithoutReturn", src: `
+    (function() {})();
+    `,
+    expected: undefined },
+
+  { name: "multipleReturn", src: `
+    var f = function() {
+        try {
+            return true;
+        }
+        finally {
+            return false;
+        }
+    }
+    f();
+    `,
+    expected: false },
+
+  { name: "throwCatch", src: `
+    var f = function () {
+        throw 26;
+    }
+    try {
+        f();
+    }
+    catch (e) {
+        e * 2;
+    }
+    `,
+    expected: 52 },
+
+  { name: "seqExpr", src: `
+    51, 52, 53;
+    `,
+    expected: 53 },
+
+  { name: "labeledStatement", src: `
+    foo: 54;
+    `,
+    expected: 54 },
+
+  { name: "whileLoop", src: `
+    var a = 0;
+    while(a<55) {
+        a++;
+    }
+    a;
+    `,
+    expected: 55 },
+
+  { name: "whileFalse", src: `
+    var a = 56;
+    while(false) {
+        a++;
+    }
+    a;
+    `,
+    expected: 56 },
+
+  { name: "doWhileFalse", src: `
+    var a = 56;
+    do {
+        a++;
+    } while(false);
+    a;
+    `,
+    expected: 57 },
+
+  { name: "breakDoWhile", src: `
+    var a = 57;
+    do {
+        a++;
+        break;
+        a++;
+    } while(false);
+    a;
+    `,
+    expected: 58 },
+
+  { name: "selfBreak", src: `
+    foo: break foo;
+    `,
+    expected: undefined /* (but legal!) */ },
+
+  { name: "breakWithFinally", src: `
+    var a = 6;
+    foo: {
+        try {
+            a *= 10;
+            break foo;
+        }
+        finally {
+            a--;
+        }
+    }
+    a;
+    `,
+    expected: 59 },
+
+  { name: "continueWithFinally", src: `
+    var a = 59;
+    do {
+      try {
+        continue;
```

Comment:
This function is JS-standard with two space indentation, but many others have 4.